packages/edit-site/src/store/actions.js

```diff
@@ -298,3 +300,140 @@ export function setIsListViewOpened( isOpen ) {
 		isOpen,
 	};
 }
+
+/**
+ * Reverts a template to its original theme-provided file.
+ *
+ * @param {Object} template The template to revert.
+ */
+export function* revertTemplate( template ) {
+	if ( ! isTemplateRevertable( template ) ) {
+		yield controls.dispatch(
+			'core/notices',
+			'createErrorNotice',
+			__( 'This template is not revertable.' ),
+			{ type: 'snackbar' }
+		);
+		return;
+	}
+
+	try {
+		const serializeBlocks = ( { blocks: blocksForSerialization = [] } ) =>
+			__unstableSerializeAndClean( blocksForSerialization );
+
+		const edited = yield controls.select(
+			'core',
+			'getEditedEntityRecord',
+			'postType',
+			'wp_template',
+			template.id
+		);
+		yield controls.dispatch(
+			'core',
+			'editEntityRecord',
+			'postType',
+			'wp_template',
+			template.id,
+			{
+				content: serializeBlocks,
```

Comment:
What are we actually doing here?

I've noticed that the revert works even without passing anything to `content` (or passing some random value).

Actually, this is the action as recorded by the Redux Devtools:

<img width="432" alt="Screenshot 2021-03-09 at 14 21 36" src="https://user-images.githubusercontent.com/2070010/110484692-cf129d80-80e2-11eb-8505-a966753afc38.png">

Neither `content` nor `is_custom` show up there. 😮

Reply:
Without `content` it works fine until you save and reload the page. The function we provide for `content` here tells the saving flow to serialize the blocks. Without `content` it will never save the actual blocks to the server.

If you comment out `content` then the last 2 tests are failing.

`is_custom` seems to be redundant though. I'll make the tests a little bit more accurate to make sure we don't need it.

Reply:
What we are actually doing here is fixing the `undo level` to contain the correct values. Without this `dispatch` the undo just doesn't work.

Reply:
Added a few checks for save button disabled state. If I remove `is_custom` then the save button remains enabled after clicking undo. The editor thinks the entity is dirty so you can press the save button. -> `is_custom` is required here 😄